[to-branch-Pluto] Sidekiq Z2: add basic support for board

commodo opened the pull request:
This changeset adds support for both variants of SidekiqZ2 (with Micron and ISSI flash devices).
The only thing that is up for discussion here is patch ( e3131d6d3673b7b07f6ddb800e07f70228bc1e6b : spi_flash: do not return error code for unsupported flash lock/unlock ops )

There is one variant for the SidekiqZ2 with ISSI flash and it doesn't support `sf protect lock/unlock`.
That's because it's only supported for Micron/STmicro devices.
Supporting it is possible, but requires more work, because ISSI needs to access another register for doing the same lock/unlock as Micron/STmicro.

The rest of the change is just a copy of the Pluto with names updated to Epiq Solutions & SidekiqZ2, as well as USB vendor + product ID updates.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

commodo committed:
arch: arm: add support for Sidekiq Z2 board

This adds support for the Sidekiq Z2 board, which has been adapted from
Pluto.

What differs from Pluto:
* all names referring to Pluto/PlutoSDR have been changed to SidekiqZ2
* USB vendor & product ID has been changed to Epiq Solutions & SidekiqZ2

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

commodo committed:
spi: spi_flash: add support for ISSI IS25WP256D flash

Datasheet is:
  http://www.issi.com/WW/pdf/IS25LP(WP)256D.pdf

There is a version of the Sidekiq Z2 with this flash chip.
This change adds support for that chip.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

commodo committed:
spi_flash: do not return error code for unsupported flash lock/unlock ops

Instead, print a warning message for the user to see.

This is, in order to support non-Micron/STMicro flashes for
internal/experimental usage, or for other boards that are similar to
Pluto/M2k, but with different flash types.

The flashing operation will try to do a `sf protect lock/unlock` on the
flash, and that will error out for non-Micron/STMicro flashes.
That's because other flash types don't support the ops [in the driver].

Disabling an error condition is not nice, but it's easier than modifying
the u-boot vars for each flash-type. And we still get a warning message, so
there should be awareness about this.

Supporting lock/unlock for ISSI flashes is possible, but requires more work
because ISSI needs to access another register for doing the same
lock/unlock as Micron/STmicro (i.e. access the Top/Bottom select & BP3
bit).

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

mhennerich commented on arch/arm/dts/zynq-sidekiqz2.dts:
```diff
@@ -0,0 +1,77 @@
+
+/ {
+	model = "Zynq Sidekiq Z2 Board";
+	compatible = "xlnx,zynq-picozed-sdr2", "xlnx,zynq-7000";
```
xlnx,zynq-picozed-sdr2 copy and paste error - pluto might need to be fixes as well

commodo replied:
ack ; will remove ;
will also check `pluto`

mhennerich commented on include/spi_flash.h:
```diff
@@ -233,8 +233,11 @@ static inline int spi_flash_erase(struct spi_flash *flash, u32 offset,
-	if (!flash->flash_lock || !flash->flash_unlock)
-		return -EOPNOTSUPP;
+	if (!flash->flash_lock || !flash->flash_unlock) {
+		printf("*** Warning: flash lock/unlock ops not supported "
```
What's the implication? The script fails?

commodo replied:
so, without this patch, u-boot does not enter DFU mode; because `sf protect` returns `-EOPNOTSUPP`
the u-boot cmd that fails [if i remember correctly] is:
```
"jtagboot=env default -a;sf probe && sf protect unlock 0 100000 && run dfu_sf; \0"
```

another approach, would be to do a different u-boot for non-STMicro/Micron flashes without `sf protect` ; or to enter DFU mode regardless of the `sf protect` return code ;
atm, there is only 1 SidekiqZ2 board with an ISSI flash;
the other one [with STMicro/Micron] works without this change ;

commodo committed:
zynq-pluto-sdr: remove `xlnx,zynq-picozed-sdr2` compat string

The Pluto is not compatible with the `xlnx,zynq-picozed-sdr2` board, so
remove this string from the compatible list.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

mhennerich commented:
ok - let's keep the scripts the same.

commodo commented:
ack ; will re-spin with changes to DTS files; 

ideally, what would be nice, is to have implemented ISSI lock/unlock ; that requires a bit of work, because some things are not common with STMicro/Micron types; 
the Top/Bottom and BP3 bits are in a different places/registers ;

but, will see later ;

commodo commented:
Changelog v1 -> v2: 
* removed  `xlnx,zynq-picozed-sdr2` string from Pluto & Sidekiq compatible list

dbogdan commented on arch/arm/dts/zynq-sidekiqz2.dts:
```diff
@@ -0,0 +1,77 @@
+/*
+ * Avnet PicoZED SDR2 board DTS
+ *
```
This is not the Avnet PicoZED SDR2 board...

commodo replied:
good catch ; will fix :)

commodo committed:
zynq-pluto-sdr.dts: correct file header and update copyright

The Pluto DTS file header also contained incorrect info about the board.
Updated to `Analog Devices Inc. Pluto SDR board DTS`
And updated copyright years.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

commodo commented:
Changelog v2 -> v3: 
* updated Pluto DTS file header info
* corrected SidekiqZ2 DTS file header info

commodo commented:
ping
